docs/overview.rst

```diff
@@ -0,0 +1,350 @@
+-----------------
+
+If you use Cython, you can't use HPy directly. The plan is to write a Cython
+backend which emits HPy code instead of Python/C code: once this is done, you
+will get the benefits of HPy automatically.
+
+
+Extensions in other languages
+------------------------------
+
+On the API side, HPy is designed with C in mind, so it is not directly useful
+if you want to write an extension in a language different than C.
+
+However, Python bindings for other languages could decide to target the HPy
+Universal ABI instead of the CPython ABI, and generate extensions which can be
+loaded seamlessly on all Python implementations which supports it.  This is
+the route taken for example by Rust (XXX put a link?).
+
+
+Benefits for alternative Python implementations
+------------------------------------------------
+
+If you are writing an alternative Python implementation, there is a good
+chance that you already know how painful it is to support the Python/C
+API. HPy is designed to be both faster and easier to implement!
+
+You have two choices:
+
+  - support the Universal ABI: in this case, you just need to export the
+    needed functions and to add a hook to ``dlopen()`` the desired libraries
+
+  - use a custom ABI: in this case, you have to write your own replacement for
+    ``hpy.h`` and recompile the C extensions with it.
+
+
+Current status and roadmap
+--------------------------
+
+At the moment of writing, HPy is still in its early stages of development. The
+following milestones have been reached:
+
+  - it is possible to write extensions which expose module-level functions,
+    with all the various kinds of calling conventions
+
+  - there is a limited support for argument parsing (only a couple of basic
+    types actually work)
+
+  - there is support for raising and catching exceptions
+
+  - it is possible to choose between the CPython and the HPy Universal ABIs
+    when compiling an extension module
+
+  - extensions compiled with the CPython ABI work out of the box on CPython
+
+  - it is possible to load HPy Universal extensions on CPython, thanks to the
+    ``hpy.universal`` package
+
+  - it is possible to load HPy Universal extensions on
+    PyPy (using the PyPy `hpy branch <https://foss.heptapod.net/pypy/pypy/tree/branch/hpy>`_
+
+  - it is possible to load HPy Universal extensions on GraalPython (XXX link)
+
+
+However, there is still a long road before HPy is usable for the general
+public. In particular, the following features are on our roadmap but has not
+been implemented yet:
+
+  - it is not possible to write custome types in C. There is already a WIP
+    branch to address this issue
+
+  - only a handful of the original Python/C functions has been ported to
+    HPy. Porting most of them is straighforward, so for now the priority is to
+    work on the "hard" features to prove that the HPy approach works, and we
+    port new functions as needed
+
+  - the debug mode simply does not exist (yet!)
```

Comment:
Link to a discussion of how to implement? Otherwise it sounds like a pie-in-the-sky idea that may or may not work

Reply:
We don't have any discussion in a written form AFAIK.
I originally plan to write also a document `debugging.rst` which explains some of the ideas, but I thought it is better to start merging this at first, to allow smaller reviews